### PR TITLE
fix(voip): Android DDP thread safety and VoipPayload bundle parity

### PR DESCRIPTION
**Pull request by diegolmello**
## Proposed changes

Android VoIP DDP and cold-start payload hardening (split-plan **PR 2**):

- **`DDPClient`**: use `AtomicInteger` for DDP message ids; mark `isConnected` and `connectedCallback` `@Volatile` for visibility across OkHttp and main-thread handlers.
- **Connect timeout**: track a single `Runnable` for the connect timeout and remove only that callback instead of `removeCallbacksAndMessages(null)`, so unrelated `Handler` work is not cleared. Cancel that timeout on `connected`, `disconnect()`, and WebSocket `onFailure`.
- **Stale-listener guard**: `onMessage`, `onFailure`, and `onClosed` now bail early when the `WebSocket` parameter does not match the current active socket (`webSocket !== this@DDPClient.webSocket`). Prevents a closed socket's late OkHttp callbacks from hijacking a newly installed `connectedCallback` via a CAS win after reconnect.
- **`disconnect()` no longer resets `connectResultDelivered`**: `connect()` already rearms the flag via `resetConnectHandshakeState()`; the extra reset in `disconnect()` narrowed the stale-listener guard window unnecessarily.
- **`VoipPayload.toBundle()`**: include `voipAcceptFailed` so process-death / bundle restore matches `fromBundle()` and `toWritableMap()`.
- **`VoipModule`**: `@Volatile` on companion `initialEventsData` for cross-thread visibility.
- **Tests**: Robolectric unit tests for connect-timeout vs `connected` ordering, `voipAcceptFailed` bundle round-trip, connection failure delivery (previously untested `onFailure` path), and stale-failure no-op regression lock.

## How to test or reproduce

From repo root after `yarn`:

```bash
cd android && ./gradlew :app:testOfficialDebugUnitTest --tests "chat.rocket.reactnative.voip.DDPClientTest" --tests "chat.rocket.reactnative.voip.VoipPayloadBundleTest"
```

Or run all VoIP JVM unit tests:

```bash
cd android && ./gradlew :app:testOfficialDebugUnitTest --tests "chat.rocket.reactnative.voip.*"
```

## Screenshots

N/A (native concurrency and serialization only).

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [ ] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Draft PR: branch `pr2-android-ddp-thread-safety`. Base branch is `feat.voip-lib-new`.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improve VoIP reliability: ensure handshake callback fires at most once, robustly handle timeouts and stale socket events, and make initial event handling and message sending concurrency-safe.
  * Include accept-failure flag in VoIP payload serialization for reliable process-recovery.

* **Tests**
  * Add Android unit tests covering handshake timing, duplicate/late messages, timeout cancellation, stale-failure regressions, and payload bundle round-trips.

* **Chores**
  * Enable Robolectric-based Android unit testing.

### CODE REVIEW

File: android/app/src/main/java/chat/rocket/reactnative/voip/DDPClient.kt [hunk 1]
```diff
@@ -134,19 +150,21 @@ class DDPClient {
...
         isConnected = false
+        cancelConnectTimeout()
         synchronized(pendingCallbacks) { pendingCallbacks.clear() }
         clearQueuedMethodCalls()
         connectedCallback = null
+        connectResultDelivered.set(false)
         onCollectionMessage = null
         webSocket?.close(1000, null)
         webSocket = null
     }
```
[the remark this answers is not in the document]
Comment (diegolmello) [reply android/app/src/main/java/chat/rocket/reactnative/voip/DDPClient.kt:168]: Good catch. Applied as a 3-part fix in the latest push:

1. **Removed `connectResultDelivered.set(false)` from `disconnect()`** — as you suggest; `connect()` already rearms via `resetConnectHandshakeState()`.
2. **Added stale-listener guard** in `onMessage`, `onFailure`, and `onClosed`: `if (webSocket !== this@DDPClient.webSocket) return`. This is the primary protection — even after your proposed fix alone, a rapid reconnect (connect2 calls resetHandshakeState → delivered=false) still leaves a window where a late L1 callback wins the CAS. The instance-identity check closes that window entirely.
3. **Added two tests**: `connection failure delivers false exactly once` (previously untested onFailure path) and `stale failure after completed connect and disconnect is a no-op` (regression lock for the CAS hijack when no new callback is armed).
